akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.internal
+
+import scala.collection.immutable
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.testkit.typed.scaladsl.SerializationTestKit
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.annotation.InternalApi
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit.CommandResultWithReply
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit.RestartResult
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit.SerializationSettings
+import akka.persistence.testkit.scaladsl.PersistenceTestKit
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.internal.EventSourcedBehaviorImpl
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object EventSourcedBehaviorTestKitImpl {
+  final case class CommandResultImpl[Command, Event, State, Reply](
+      command: Command,
+      events: immutable.Seq[Event],
+      state: State,
+      replyOption: Option[Reply])
+      extends CommandResultWithReply[Command, Event, State, Reply] {
+
+    override def isNoEvents: Boolean = events.isEmpty
```

Comment:
`hasNoEvents`? 

Could also be paired with `hasNoReply`

Reply:
yes, hasNoEvents is better.

It's is always expected to be a reply when the `runCommand(ActorRef => Command)` is used. Otherwise it will throw AssertionError from runCommand.

For more advanced reply later cases the other `runCommand(Command)` can be used with an ordinary TestProbe, but that should be a rather rare scenario.